skill-tree/src/main/java/com/RDS/skilltree/services/SkillServiceImplementation.java

```diff
@@ -30,6 +37,71 @@ public List<SkillViewModel> getAll() {
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public SkillRequestsDto getAllRequests() {
```

Comment:
1. Move the logic for fetching user details outside the map and forEach loop.
2. Create a set to store user IDs for whom details have already been fetched.
3. Before fetching details using rdsService.getUserDetails, check if the user ID is already present in the set.
4. If the ID is not present in the set, fetch the details, add the ID to the set and store the details in the userDetails map.